src/test/java/info/sjd/controller/UserControllerTest.java

```diff
@@ -0,0 +1,157 @@
+        ResponseEntity<User> responseEntity = testRestTemplate.exchange(requestEntity, User.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(userService, times(1)).findById(anyInt());
+    }
+
+    @Test
+    void getAllTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        when(userService.findAll()).thenReturn(Collections.singletonList(user));
+        RequestEntity<User> requestEntity = new RequestEntity<>(new User(), HttpMethod.GET, new URI("/user"));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(userService, times(1)).findAll();
+    }
+
+    @Test
+    void getByLoginTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        when(userService.findByLogin(anyString())).thenReturn(user);
+        UriComponents uriComponents = UriComponentsBuilder.newInstance()
+                .host("/user")
+                .path("/by-login").query("login={keyword}").buildAndExpand("log");
+        RequestEntity<User> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<User> responseEntity = testRestTemplate.exchange(requestEntity, User.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(userService, times(1)).findByLogin(anyString());
+    }
+
+    @Test
+    void getByLoginAndPasswordTest() throws URISyntaxException {
+        User user = User.builder().
+                login("test_login111").
+                password("test_pass").
+                firstName("test_fn").
+                lastName("test_ln").
+                build();
+        when(userService.findByLoginAndPassword(anyString(), anyString())).thenReturn(user);
+        UriComponents uriComponents = UriComponentsBuilder.newInstance()
+                .host("/user")
+                .path("by-login-and-password").query("login={keyword}").query("password={keyword}").buildAndExpand("log", "pass");
+        RequestEntity<User> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<User> responseEntity = testRestTemplate.exchange(requestEntity, User.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(userService, times(1)).findByLoginAndPassword(anyString(), anyString());
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
```

Comment:
The same situation as previous. You must mock and check delete method

Reply:
done